Stop motors when stabilizer task is blocking

gemenerik opened the pull request:
This PR introduces a new task, `rateSupervisorTask`, to monitor the `stabilizerTask` and detect if it is blocking. The `rateSupervisorTask` uses a semaphore to synchronize with the `stabilizerTask` and validate the loop rate. If the `stabilizerTask` is blocking, the `rateSupervisorTask` will stop the motors. This supervisor task also takes over the existing rate supervising that was previously done within the `stabilizerTask`. Fixes #1406.

Tested by sleeping in the supervisor task. 

We should discuss after how long this should step in. Currently at 2000 ms, but I think we could intervene more quickly.

ToveRumar reviewed (commented):
It looks like for example config.h has things that are not really changes checked in. Might be line ending differences?

ToveRumar commented:
> Perhaps it would be even better to just ASSERT when this happens?

Agree. I think asserting is a good idea in this case.

gemenerik commented:
> It looks like for example config.h has things that are not really changes checked in. Might be line ending differences?

I added `RATE_SUPERVISOR_TASK_PRI`, `RATE_SUPERVISOR_TASK_NAME`, `RATE_SUPERVISOR_TASK_STACKSIZE` and adjusted indentation of others to match the longer name. Diff doesn't seem to pick up on it nicely.

gemenerik commented:
> > Perhaps it would be even better to just ASSERT when this happens?
> 
> Agree. I think asserting is a good idea in this case.

With latest commit:

```
SYS: The system resumed after a failed assert [WARNING]
SYS: Assert failed at ../src/modules/src/stabilizer.c:274
```
